SteamKit2/SteamKit2/Util/HardwareUtils.cs

```diff
@@ -281,12 +272,63 @@ public override byte[] GetDiskId()
         }
     }
 
-    static class HardwareUtils
+    /// <summary>
+    /// Class to allow user to provide custom machine info provider.
+    /// </summary>
+    public static class HardwareUtils
     {
+        private static IMachineInfoProvider userMachineInfoProvider;
+        private static bool IsMachineInfoProviderInitialized;
+
+        /// <summary>
+        /// User-provided MachineInfoProvider.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Occurs when a user tries to (re-)set this property after its first usage.</exception>
+        public static IMachineInfoProvider UserMachineInfoProvider
+        {
+            get => userMachineInfoProvider;
```

Comment:
Why do we need a property for a "user-provided machine info provider", rather than just having a property for a "machine info provider"? It seems odd that a user can't just get the provider themselves.